jax/_src/prng.py

```diff
@@ -216,9 +218,17 @@ def delete(self) -> None: ...
   def is_deleted(self) -> bool: ...
   @abc.abstractmethod
   def on_device_size_in_bytes(self) -> int: ...
+  @property
+  @abc.abstractmethod
+  def addressable_shards(self) -> List[Shard]: ...
+  @property
+  @abc.abstractmethod
+  def global_shards(self) -> List[Shard]: ...
+  @abc.abstractmethod
+  def addressable_data(self, index: int) -> PRNGKeyArray: ...
```

Comment:
Relatedly, but not necessarily for this PR: should we have (and inherit) some basic "jax array" interface, which would require methods like these?

Reply:
If we ever find ourselves doing this a second time, we should add that abstraction!